src/bestcrypt_ve_fmt_plug.c

```diff
@@ -0,0 +1,456 @@
+/*
+ * JtR format to crack password protected BestCrypt v4 volume
+ * This implementation relies on findings from @trounce1 and @kholia 
+ * reverse-engineering work
+
+ * ----------------------------------------------------------------------------
+ * "THE BEER-WARE LICENSE" (Revision 42):
+ * <jean-christophe.delaunay@synacktiv.com> wrote this file. As long as you 
+ * retain this notice you can do whatever you want with this stuff. If we meet 
+ * some day, and you think this stuff is worth it, you can buy me a beer in 
+ * return
+ * ----------------------------------------------------------------------------
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_bestcrypt_ve;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_bestcrypt_ve);
+#else
+
+#include <string.h>
+#include <errno.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "yescrypt/yescrypt.h"
+#include "aes.h"
+#include "twofish.h"
+#include "serpent.h"
+#include <openssl/camellia.h>
```

Comment:
Now that we've made OpenSSL optional, this format should also exclude either itself completely or its support for Camellia when building `--without-openssl` - please refer to recent changes to other formats for how to do this.

Reply:
Configuring with flags `--without-openssl --with-commoncrypto` leads to a compilation error in another format:

```
aram allow-store-data-races=0 -Wno-deprecated-declarations -Wunused-but-set-variable -std=gnu89 -Wdate-time -D_POSIX_SOURCE -D_GNU_SOURCE -D_XOPEN_SOURCE=600  -fopenmp  -pthread -I/usr/local/include -DCL_SILENCE_DEPRECATION -DHAVE_OPENCL -pthread -funroll-loops gpg_common_plug.c -o gpg_common_plug.o
gpg_common_plug.c: In function ‘check_dsa_secret_key’:
gpg_common_plug.c:1086:29: error: dereferencing pointer to incomplete type ‘DSA’ {aka ‘struct dsa_st’}
  error = BN_mod_exp(res, dsa->g, dsa->priv_key, dsa->p, ctx);
                             ^~
gpg_common_plug.c: In function ‘gpg_common_check’:
gpg_common_plug.c:1595:9: error: storage size of ‘dsa’ isn’t known
     DSA dsa;
         ^~~
gpg_common_plug.c:1595:9: warning: unused variable ‘dsa’ [-Wunused-variable]
```

Reply:
We no longer support `--with-commoncrypto`, see ad47c9c1607ed870c018a904694519f6f0ad613d. Why did you pass this option? Where did you even find it documented? If we somehow didn't drop it fully enough, then we should complete that.

Please try configuring/building without that option (don't forget `make clean` after reconfiguring), and let us know whether the build errors remain or are gone. Thanks!

Reply:
Indeed, the sources I was relying on were too old. I don't have the problem anymore (`--with-commoncrypto` doesn't exist anymore)